draft-ietf-tls-tls13.md

```diff
@@ -4215,30 +4215,28 @@ user_canceled
   SHOULD be followed by a "close_notify". This alert is generally a warning.
 {:br }
 
-Either party MAY initiate a close by sending a "close_notify" alert. Any data
-received after a closure alert MUST be ignored. If a transport-level close is
-received prior to a "close_notify", the receiver cannot know that all the
-data that was sent has been received.
+Either party MAY initiate a close of its write side of the connection by
+sending a "close_notify" alert. Any data received after a closure alert MUST
+be ignored. If a transport-level close is received prior to a "close_notify",
+the receiver cannot know that all the data that was sent has been received.
 
-Each party MUST send a "close_notify" alert before closing the write side
-of the connection, unless some other fatal alert has been transmitted. The
-other party MUST respond with a "close_notify" alert of its own and close down
-the connection immediately, discarding any pending writes. The initiator of the
-close need not wait for the responding "close_notify" alert before closing the
-read side of the connection.
+Each party MUST send a "close_notify" alert before closing its write side
+of the connection, unless it has already sent some other fatal alert. This
+does not have any effect on its read side of the connection. The other party
+SHOULD NOT react to a "close_notify" by discarding pending writes and
+sending an immediate "close_notify" alert of its own, as that could cause
+truncation in the read side. Both parties need not wait to receive a
+"close_notify" alert before closing their read side of the connection.
```

Comment:
I think this is too much a response to the original text. I would just remove "The other party SHOULD NOT" and instead say "Note that this is a change from versions of TLS prior to TLS 1.3 in which implementations were required to immediately respond to a close_notify with a close_notify of their own"

Reply:
I removed the SHOULD NOT and added the note but kept the mention of risks of truncation